---
title: "stable_LPS-picrust2-aldex2"
author: "Mike Connelly"
date: "4/2/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/projects/EAPSI_Pocillopora_LPS")
```
## Setup packages and working directories
```{r, message=FALSE}
library(ALDEx2)
library(phyloseq)
library(microbiome)
library(tidyverse)
library(patchwork)
```

Setup sample metadata for ALDEx2 testing
```{r}
map <- read.table("./data/qiime2_metadata.tsv", sep ="\t", row.names = 1, header = FALSE)
colnames(map) <- c ("Reef",	"Colony",	"Environment",	"Treatment")
map$SampleID <- rownames(map)
map$Reef <- factor(map$Reef, levels = c("Houwan", "Wanglitung", "controls"), ordered = TRUE)
map$Colony <- factor(map$Colony, levels = c("HW1", "HW2", "WT1", "WT2", "controls"), ordered = TRUE)
map$Treatment <- factor(map$Treatment, levels = c("control", "LPS", "positive", "negative"), ordered = TRUE)
```

Read in predicted metagenome pathway abundance tables from PICRUSt2 results
```{r}
ec_predicted <- read.delim("./outputs/picrust2_out_pipeline/EC_predicted.tsv", header = TRUE)
ko_predicted <- read.delim("./outputs/picrust2_out_pipeline/EC_predicted.tsv", header = TRUE)
path_abund_unstrat <- read.delim("./outputs/picrust2_out_pipeline/pathways_out/path_abun_unstrat.tsv", header = TRUE)
```
```{r}
path_abund <- path_abund_unstrat %>% dplyr::select(pathway, matches("[HW][wt][1-2].[36][abc]"))
row.names(path_abund) <- path_abund$pathway
path_abund <- path_abund %>% dplyr::select(-pathway)
path_abund_round <- round(path_abund, digits = 0)
```
```{r}
sum(path_abund)
relative <- function(x) {round(((x / sum(x))*9500000), digits = 0)}
path_abund_rel <- relative(path_abund)
sum(path_abund_rel)
```
```{r}
aldex.asv <- path_abund_rel
```
```{r}
conds <- as.character(sample_data(ps.LPS)$Treatment)
conds
```

## ALDEx2 tests
```{r}
x.ps <- aldex(reads = aldex.asv, conditions = conds, mc.samples = 128, test = "t", effect = TRUE, include.sample.summary = FALSE, denom = "all", verbose = TRUE)
```
```{r}
write.csv(x.ps, file = "./outputs/aldex2/lps-picrust2-aldex2.csv")
```
```{r}
pdf("./outputs/aldex2/picrust2_out_pipeline/picrust2-aldex2.pdf", height = 6, width = 10)
par(mfrow=c(1,2))
aldex.plot(x.ps, type = "MA", test = "welch")
aldex.plot(x.ps, type="MW", test="welch")
dev.off()
```

```{r, ggplot aldex.plot function}
aldex.plot_gg <- function (x, ..., type = c("MW", "MA"), xlab = NULL, ylab = NULL, 
    xlim = NULL, ylim = NULL,
    rare = 0, cutoff = 0.1,
    all.col = rgb(0, 0, 0, 0.2), called.col = "red", rare.col = "black",
    pointsize = 3,
    thres.line.col = "darkgrey", test = "welch") 
{
    type <- match.arg(type)
    if (length(x$effect) == 0) 
        stop("Please run aldex.effect before plotting")
    if (test == "welch") {
        if (length(x$we.eBH) == 0) 
            stop("Welch's t test results not in dataset")
        called <- x$we.eBH <= cutoff
    }
    else if (test == "wilcox") {
        if (length(x$wi.eBH) == 0) 
            stop("Wilcoxon test results not in dataset")
        called <- x$wi.eBH <= cutoff
    }
    if (test == "glm") {
        if (length(x$glm.eBH) == 0) 
            stop("glm test results not in dataset")
        called <- x$glm.eBH <= cutoff
    }
    if (test == "kruskal") {
        if (length(x$kw.eBH) == 0) 
            stop("Kruskall-Wallace test results not in dataset")
        called <- x$kw.eBH <= cutoff
    }
    if (type == "MW") {
        if (is.null(xlab)) 
            xlabel <- expression("Median" ~ ~Log[2] ~ ~"win-Condition diff")
        if (is.null(ylab)) 
            ylabel <- expression("Median" ~ ~Log[2] ~ ~"btw-Condition diff")
        x$asv.type <- ifelse(x$rab.all < rare, "rare", ifelse(x$we.eBH < cutoff, "called", "all"))
        ggaldex_mw <- ggplot(data = x, aes(x = diff.win, y = diff.btw, color = asv.type)) +
          geom_point(size = pointsize) +
          geom_abline(aes(intercept = 0, slope = 1), linetype = "dashed", color = thres.line.col) +
          geom_abline(aes(intercept = 0, slope = -1), linetype = "dashed", color = thres.line.col) +
          scale_x_continuous() +
          scale_y_continuous() +
          scale_color_manual(name = "ASV Type", values = c(all.col, rare.col, called.col)) +
          xlab(xlabel) +
          ylab(ylabel)
        return(ggaldex_mw)
    }
    if (type == "MA") {
        if (is.null(xlab)) 
            xlabel <- expression("Median" ~ ~Log[2] ~ ~"relative abundance")
        if (is.null(ylab)) 
            ylabel <- expression("Median" ~ ~Log[2] ~ ~"btw-Condition diff")
        x$asv.type <- ifelse(x$rab.all < rare, "rare", ifelse(x$we.eBH < cutoff, "called", "all"))
        ggaldex_ma <- ggplot(x, aes(x = rab.all, y = diff.btw, color = asv.type)) +
          geom_point(size = pointsize) +
          scale_x_continuous() +
          scale_y_continuous() +
          scale_color_manual(name = "ASV Type", values = c(all.col, rare.col, called.col)) +
          xlab(xlabel) +
          ylab(ylabel)
        return(ggaldex_ma)
    }
}
```
```{r}
aldex1 <- aldex.plot_gg(x.ps, type = "MW", test = "welch")
aldex1 <- aldex1 + ggtitle(subtitle = "LPS vs. Control Treatment", "Bacteria PICRUSt2-predicted functional gene content ALDeX2 Differential Abundance MW/MA Plots") 

aldex2 <- aldex.plot_gg(x.ps, type = "MA", test = "welch")
 
aldexplots <- (aldex1 / aldex2) 
aldexplots
ggsave(aldexplots, filename = "./outputs/aldex2/lps-picrust2-aldex2.pdf", device = "pdf", width = 8.5, height = 5.5)
```
```{r}
cutoff <- 0.1
called <- x.ps[x.ps$we.eBH <= cutoff, ]
called.ASVs <- taxonomy[rownames(called), ]
called.ASVs1 <- cbind(called, called.ASVs)
```
```{r}
rare_cutoff <- 0
rare <- x.ps[x.ps$rab.all < rare_cutoff, ]
rare.ASVs <- taxonomy[rownames(rare), ]
rare.ASVs1 <- cbind(called, rare.ASVs)
```
```{r}
?aldex.clr
?aldex.plot
?aldex.effect
```
