---
title: "LPS_WGCNA-dendrograms_Pdam"
author: "Mike Connelly"
date: "11/05/2019"
output: html_document
---

RMarkdown document containing scripts for WGCNA modelling of differential gene expression between fragments of two *Pocillopora damicornis* genotypes collected at Wanglitung reef and placed in five experimental treatment groups and once control group during EAPSI Taiwan, summer 2017.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/scripts/EAPSI_HW-WT-master/")
```
## Load packages and setup working directory
```{r, error=FALSE, warning=FALSE, message=FALSE}
library(tidyverse)
library(RColorBrewer)
library(genefilter)
library(WGCNA)
library(dynamicTreeCut)
library(flashClust)
options(stringsAsFactors = FALSE)
allowWGCNAThreads(nThreads = 4)
```

After network construction and module eigengene calculation...

## Plot the dendrogram and the module colors underneath
```{r}
geneTree <- net$dendrograms[[1]]
```
```{r}
plotDendroAndColors(dendro = net$dendrograms[[1]], colors = moduleColors[net$blockGenes[[1]]],
"Module Colors",
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.15,
abHeight = c(0.95))
```
## Visualize eigengene network
```{r}
plotEigengeneNetworks(MEs, "Eigengene dendrogram", marDendro = c(0,4,2,0),
plotHeatmaps = FALSE, excludeGrey = TRUE)
```
```{r}
plotEigengeneNetworks(MEs, "Eigengene adjacency heatmap", marHeatmap = c(3,4,2,2),
plotDendrograms = FALSE, colorLabels = TRUE, xLabelsAngle = 90, excludeGrey = TRUE)
```
```{r}
plotEigengeneNetworks(MEs1, "Eigengene dendrogram", marDendro = c(0,4,2,0),
plotHeatmaps = FALSE, colorLabels = TRUE, excludeGrey = TRUE)
```
```{r}
plotEigengeneNetworks(MEs1, "Eigengene adjacency heatmap", marHeatmap = c(3,4,2,2),
plotDendrograms = FALSE, xLabelsAngle = 90)
```

## Topological Overlap Map
```{r}
load("EAPSI_TOM-block.1.RData")
```
```{r}
dissTOM <- 1 - as.matrix(TOM)
```
```{r}
# Transform dissTOM with a power to make moderately strong connections more visible in the heatmap
plotTOM <- dissTOM^7;
# Set diagonal to NA for a nicer plot
diag(plotTOM) = NA;
```
```{r}
rm(dissTOM)
```
```{r}
TOMplot(dissim = plotTOM, dendro = geneTree, Colors = moduleColors, setLayout = TRUE, main = "Network heatmap plot, all genes")
```
