---
title: "Stable_DESeq-boxplots_Pdam"
author: "Mike Connelly"
date: "10/24/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/scripts/EAPSI_HW-WT-master/")
```
## Setup packages and working directories
```{r packages, error=FALSE, warning=FALSE, message=FALSE}
library("DESeq2")
library("tidyverse")
library("ggplot2")
library("ggrepel")
library("RColorBrewer")
library("scales")
library("ggpubr")
library("grid")
```
Set colors and shapes for ggplot options
```{r}
theme_set(theme_bw())
colshapes <- c(16, 17, 15, 18)
condcolors_LPS <- c("#0000FF", "#FF66FF")
colcolors <- c("olivedrab3", "springgreen", "deepskyblue", "skyblue")
condfillcolors_LPS <- c("#99CCFF", "#FFCCFF")
```

Create ggboxplot functions
```{r}
ggboxplot <- function(gene, res, samples) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds_LPS, gene = gene, intgroup = "Treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=Treatment, y=count, color=samples$Treatment)) 
      + geom_boxplot(aes(fill=samples$Treatment), show.legend = FALSE) 
      + geom_point(aes(shape = samples$Colony), size = 3, show.legend = FALSE) 
      + scale_color_manual(values=condcolors_LPS) + scale_fill_manual(values = condfillcolors_LPS) + scale_shape_manual(values = colshapes) + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
```
```{r}
ggboxplot_geno <- function(gene, res, samples) {
titlegene <- grep(gene, res$ID)
plotTitle <- res$GeneInfo[titlegene]
gplot <- plotCounts(dds_LPS, gene = gene, intgroup = "Treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=Treatment, y=(count+0.5), color=samples$Treatment)) 
      + geom_boxplot(aes(fill=samples$Treatment), show.legend = FALSE) 
      + geom_point(aes(shape = samples$Colony), size = 3, show.legend = FALSE) 
      + facet_grid(.~samples$Colony)
      + scale_color_manual(values=condcolors_LPS) + scale_fill_manual(values = condfillcolors_LPS) + scale_shape_manual(values = colshapes) 
      + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
```
```{r}
ggboxplot_geno("pdam_00002773", res_LPS, stable_samples_LPS)
ggboxplot("pdam_00023321", res_LPS, stable_samples_LPS)
ggboxplot("pdam_00016279", res_LPS, stable_samples_LPS)
```


```{r}
gglistplot <- function(genelist, dds, res, samples) {
for (gene in genelist) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "Treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=Treatment, y=count, color=samples$Treatment)) 
      + geom_boxplot(aes(fill=samples$Treatment), show.legend = FALSE) 
      + geom_point(aes(shape = samples$Colony), size = 3, show.legend = FALSE) 
      + scale_color_manual(values=condcolors_LPS) + scale_fill_manual(values = condfillcolors_LPS) + scale_shape_manual(values = colshapes) + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
}
```
```{r}
gglistplot_genotype <- function(genelist, dds, res, samples) {
for (gene in genelist) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "Treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=Treatment, y=count, color=samples$Treatment)) 
      + geom_boxplot(aes(fill=samples$Colony), show.legend = FALSE) 
      + geom_point(aes(shape = samples$Colony), size = 3, show.legend = FALSE) 
      + facet_grid(.~samples$Colony)
      + scale_color_manual(values=condcolors_LPS) + scale_fill_manual(values = colcolors) + scale_shape_manual(values = colshapes) + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
}
```

Edit gene name
```{r}
res_LPS$IDGeneInfo <- gsub( "pdam_00002776 unknown function", "pdam_00002776 Basic helix-loop-helix family member E40", res_LPS$IDGeneInfo)
res_LPS$IDGeneInfo <- gsub( "pdam_00002776 Basic helix-loop-helix family member E40 function", "pdam_00002776 Basic helix-loop-helix family member E40", res_LPS$IDGeneInfo)
```

for use in LPS paper with proper theme and formatting
```{r}
gglistplot_genotype <- function(genelist, dds, res, samples) {
for (gene in genelist) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- rownames(res)[titlegene]
plotsub <- gsub("pdam_[0123456789]{8} ", "", res$IDGeneInfo[titlegene])
###
gplot <- plotCounts(dds, gene = gene, intgroup = "Treatment", returnData = TRUE)
gbplot <- ggplot(gplot, aes(x=Treatment, y=count, color=samples$Treatment)) +
  geom_boxplot(aes(fill = samples$Treatment), show.legend = FALSE) +
  geom_point(aes(shape = samples$Colony), size = 3, show.legend = FALSE) +
  facet_grid(.~samples$Colony) +
  scale_color_manual(values=condcolors_LPS) + scale_fill_manual(values = condfillcolors_LPS) + scale_shape_manual(values = colshapes) +
  scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x), labels = trans_format("log2", math_format(2^.x)))  + 
  ggtitle(plotTitle, subtitle = plotsub) +
  ylab("gene count") +
  theme(plot.title = element_text(size = 10, face = "bold"),
        plot.subtitle = element_text(size = 6),
        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(angle = 0, size = 10),
        strip.text = element_text(size = 10),
        strip.background = element_rect(fill = "light grey"))
print(gbplot)
###
g_bplot <- ggplot_gtable(ggplot_build(gbplot))
strip_both <- which(grepl('strip-', g_bplot$layout$name))
k <- 1
for (i in strip_both) {
j <- which(grepl('rect', g_bplot$grobs[[i]]$grobs[[1]]$childrenOrder))
g_bplot$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- colcolors[k]
k <- k+1
}
gbp <- grid.draw(g_bplot)
print(gbp)
}
}
```



## Prepare lists of gene IDs based on interesting groups
## Differentially expressed gene lists
```{r}
DElists <- list.files(path = "../EAPSI_LPS-master/DESeqresults_Pdam/lists", pattern = "_ID.txt", full.names = TRUE)
```
```{r, message=FALSE}
LPS_ctrl_up_IDs <- scan("../EAPSI_LPS-master/DESeqresults_Pdam/lists/res_LPS_ctrl_up_ID.txt", what = "character")
LPS_ctrl_dn_IDs <- scan("../EAPSI_LPS-master/DESeqresults_Pdam/lists/res_LPS_ctrl_dn_ID.txt", what = "character")
```
## Plot gene count boxplots
### DE Genes
```{r}
#pdf(file = "~/computing/scripts/EAPSI_LPS-master/DESeqresults_Pdam/figures/LPSboxplots_upDEGs.pdf", width = 8, height = 5)
gglistplot(LPS_ctrl_up_IDs, dds_LPS, res_LPS, stable_samples_LPS)
```

```{r}
pdf(file = "~/computing/scripts/EAPSI_LPS-master/DESeqresults_Pdam/figures/LPSboxplots_geno_upDEGs.pdf", width = 2.8, height = 2.1)
gglistplot_genotype(LPS_ctrl_up_IDs, dds_LPS, res_LPS, stable_samples_LPS)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_LPS-master/DESeqresults_Pdam/figures/LPSboxplots_HW1_unique.pdf", width = 2.8, height = 2.1)
gglistplot_genotype(HW1_unique, dds_LPS, res_LPS, stable_samples_LPS)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_LPS-master/DESeqresults_Pdam/figures/LPSboxplots_HW2_unique.pdf", width = 2.8, height = 2.1)
gglistplot_genotype(HW2_unique, dds_LPS, res_LPS, stable_samples_LPS)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_LPS-master/DESeqresults_Pdam/figures/LPSboxplots_WT1_unique.pdf", width = 2.8, height = 2.1)
gglistplot_genotype(WT1_unique, dds_LPS, res_LPS, stable_samples_LPS)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_LPS-master/DESeqresults_Pdam/figures/LPSboxplots_WT2_unique.pdf", width = 2.8, height = 2.1)
gglistplot_genotype(WT2_unique, dds_LPS, res_LPS, stable_samples_LPS)
```
### Innate immune genes
Lists of gene IDs
#### Pattern recognition receptors
```{r}
TLR_IDs <- scan("./genefamilies_Pdam/TLRs_IDs.txt", what = "character")
Lectin_IDs <- scan("./genefamilies_Pdam/Lectins_IDs.txt", what = "character")
#NLR_IDs
TNFR_IDs <- scan("./genefamilies_Pdam/TNFRs_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(TNFR_IDs)
```
```{r}
gglistplot(Lectin_IDs)
```

#### Signaling pathways
NFKB
```{r}
NFKB_IDs <- scan("~/computing/sequences/genomes/coral/pocillopora/pdam/genefamilies_Pdam/NFKB_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(NFKB_IDs,  dds_LPS, res_LPS, stable_samples_LPS)
```
JNK
MAPK
```{r}
TRAF_IDs <- scan("./genefamilies_Pdam/TRAFs_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(TRAF_IDs)
```

#### Effector mechanisms
MPEG1 pdam_00017055
Damicornin pdam_00023321
DELTA-stichotoxin-She4b
Phospholipases
```{r}
PL_IDs <- scan("~/computing/sequences/genomes/coral/pocillopora/pdam/genefamilies_Pdam/PL_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(PL_IDs, dds_LPS, res_LPS, stable_samples_LPS)
```
Toxins
```{r}
Tox_IDs <- scan("~/computing/sequences/genomes/coral/pocillopora/pdam/genefamilies_Pdam/Toxin_IDs.txt", what = "character")
pdf(file = "~/computing/scripts/EAPSI_LPS-master/DESeqresults_Pdam/figures/LPSboxplots_toxins.pdf", width = 8, height = 5)
gglistplot(Tox_IDs[c(3,5,6,7,9,10,11,12,13,17,18,19,20,21,22,23,24,25,26,27,28)], dds_LPS, res_LPS, stable_samples_LPS)
```
```{r}
effectorIDs <- c("pdam_00017055", "pdam_00004721", "pdam_00023321")
gglistplot(effectorIDs, dds_LPS, res_LPS, stable_samples_LPS)
```

#### Misc. immune genes
### Apoptosis pathway genes
### Antioxidant response
### Transcription factors

WGCNA Hub Genes
```{r}
hubIDs <- scan("./WGCNAmodules_Pdam_ctrl-LPS/hubgenes/hubs_IDs.txt", what = "character")
```
```{r}
gglistplot(hubIDs[1:20])
```

WGCNA Module Genes
## Input lists of co-expressed module genes after WGCNA analysis
```{r, warning = FALSE}
modulelists <- list.files(path = "./WGCNAmodules_Pdam_ctrl-LPS/lists", pattern = "_ID.txt", full.names = TRUE)
```
```{r}
blackIDs <- scan("./WGCNAmodules_Pdam_ctrl-LPS/lists/black_ID.txt", what = "character")
```
```{r}
intblackIDs <- c(14,21,29,199,200,213,231,336,359:364,402,408,415,437,481)
gglistplot(blackIDs[intblackIDs], dds_LPS, res_LPS, stable_samples_LPS)
```