---
title: "Stable_WGCNA-correlations_Pdam"
author: "Mike Connelly"
date: "10/25/2018"
output: html_document
---

RMarkdown document containing scripts for WGCNA modelling of differential gene expression between fragments of two *Pocillopora damicornis* genotypes collected at Wanglitung reef and placed in five experimental treatment groups and once control group during EAPSI Taiwan, summer 2017.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/scripts/EAPSI.HW-WT-master/")
```
## Load packages and setup working directory
```{r, error=FALSE, warning=FALSE, message=FALSE}
library(tidyverse)
library(RColorBrewer)
library(ggrepel)
library(genefilter)
library(WGCNA)
library(dynamicTreeCut)
library(flashClust)
options(stringsAsFactors = FALSE)
allowWGCNAThreads(nThreads = 4)
```
## Quantify module-trait associations
```{r}
datTraits <- dplyr::select(stable_samples, -Reef, -Colony, -Environment, -Treatment, -Batch, -heat, -anti, -anti.heat, -anti.heat.lps, - allheat, -allanti) 
rownames(datTraits) <- rownames(stable_samples)
#Define numbers of genes and samples
nGenes <- ncol(datExpr);
nSamples <- nrow(datExpr);
```
## Correlate module eigengene-trait associations
```{r}
moduleTraitCor <- cor(MEs, datTraits, use = "p")
moduleTraitPvalue <- corPvalueStudent(moduleTraitCor, nSamples)
moduleCors <- data.frame(cbind(moduleTraitCor, moduleTraitPvalue))
row.names(moduleCors) <- names(MEs)
```
## Filter significant LPS -correlated modules
```{r}
modP <- as.data.frame(moduleTraitPvalue)
modP$module <- row.names(modP)
modP <- modP[, c(10,9,8,7,6,5,4,3,2,1)]
modP <- modP %>% arrange(modP$module) %>% dplyr::select(module, lps)
colnames(modP) <- c("module", "LPSp")
```
```{r}
modPsig <- modP %>% filter(LPSp < 0.05)
sigModules <- modPsig$module
sigModules
```
## Add correlations to significant modules
```{r}
modC <- as.data.frame(moduleTraitCor)
modC$module <- row.names(modC)
modC <- modC[, c(10,9,8,7,6,5,4,3,2,1)]
modC <-  modC %>% arrange(modC$module) %>% dplyr::select(module, lps)
colnames(modC) <- c("module", "LPScor")
modCsig <- modC %>% filter(module %in% sigModules) 

modPpos <- modC %>% filter(module %in% sigModules & LPScor > 0) %>% arrange(desc(LPScor))
posModules <- modPpos$module
posModules
modPneg <-  modC %>% filter(module %in% sigModules & LPScor < 0) %>% arrange((LPScor))
negModules <- modPneg$module
negModules
```
## Get number of genes in significant modules
```{r}
numberstable$MEmodule <- paste("ME", numberstable$module, sep = "")
modN <- numberstable %>% dplyr::select(-module)
modN <- modN[,c(2,1)]
colnames(modN) <- c("module", "genes")
modN <- arrange(modN, module)
modNsig <- modN %>% filter(module %in% sigModules)
```
## Create dataframe with p-value, correlation, and number of genes
```{r}
modCPN <- as.data.frame(cbind(modP$module, modC$LPScor, modP$LPSp, modN$genes))
colnames(modCPN) <- c("module", "LPScor", "LPSp", "genes")
modCPN$LPScor <- as.numeric(modCPN$LPScor)
modCPN$LPSp <- as.numeric(modCPN$LPSp)
modCPN$genes <- as.numeric(modCPN$genes)
modCPN$correlation <- ifelse(modCPN$LPSp < 0.05 & modCPN$LPScor > 0, "Positive", ifelse(modCPN$LPSp < 0.05 & modCPN$LPScor < 0, "Negative", "NA"))
modCPN <- modCPN %>% dplyr::arrange(LPSp)
modCPN$module <- gsub("ME", "M", modCPN$module)
modCPNsig <- modCPN %>% filter(LPSp < 0.05) %>% arrange(LPSp)
```
```{r}
modCPN <- modCPN[,c(1,4,2,3,5)]
modCPN <- modCPN %>% dplyr::arrange(LPSp)
write.csv(modCPN, file = "~/computing/scripts/EAPSI_LPS-master/WGCNAmodules_Pdam_ctrl-LPS/modules.csv")
```

## Module "Volcano" Plot
```{r}
pdf(file = "~/computing/scripts/EAPSI_LPS-master/WGCNAmodules_Pdam_ctrl-LPS/figures/ModuleVolcano_LPS_8.5_labelled.pdf", width = 8.5, height = 2.5)
gVmod <- ggplot(modCPN, aes(LPScor, -log10(LPSp))) + geom_point(aes(size = genes, color = correlation))
gVmod + 
  scale_size_continuous(breaks = c(50, 500, 1000), name = "Number of Genes") +
  scale_color_manual(values = c("black", "blue", "red"), name = "LPS Trait Correlation") +
  ylim(0,5) + xlim (-1, 1) +
  geom_text_repel(data = modCPN[1:14, ], aes(label = modCPN$module[1:14]), color = "black", size = 3.5, box.padding = unit(0.2, "lines")) +
  theme_bw() +
  ggtitle("WGCNA Module Volcano Plot") +
  theme(plot.title = element_text(size = 18), axis.title = element_text(size = 14), legend.title = element_text(size = 12), axis.text = element_text(size = 12), legend.text = element_text(size = 10))
```

## Create ME-trait heatmap
```{r}
#setwd("./WGCNAmodules_Pdam_ctrl-LPS/figures")
#pdf("ModuleCorrelations_ctrl-LPS", width = 11, height = 16)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
signif(moduleTraitPvalue, 2), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor,
xLabels = names(datTraits),
yLabels = names(MEs1),
yLabelsPosition = "left",
ySymbols = names(MEs1),
yColorWidth = 0.005,
colorLabels = FALSE,
colors = blueWhiteRed(50),
textMatrix = textMatrix,
setStdMargins = FALSE,
plotLegend = FALSE,
cex.text = 0.5,
zlim = c(-1,1),
main = paste("Module-treatment correlations"))
#dev.off()
```
## Create ME-trait heatmaps for interesting modules only
```{r}
for (module in sigModules)
{
which.module <- module
intMEs <- MEs1[, which.module]
intmoduleTraitCor <- cor(intMEs, datTraits, use = "p")
intmoduleTraitPvalue <- corPvalueStudent(intmoduleTraitCor, nSamples)
# Will display correlations and their p-values
textMatrix = paste(signif(intmoduleTraitCor, 2), "\n(",
signif(intmoduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(intmoduleTraitCor)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = intmoduleTraitCor,
xLabels = names(datTraits),
yLabels = names(intMEs),
yLabelsPosition = "left",
ySymbols = names(intMEs),
yColorWidth = 0.005,
colorLabels = FALSE,
colors = blueWhiteRed(50),
textMatrix = textMatrix,
setStdMargins = FALSE,
plotLegend = FALSE,
cex.text = 0.6,
zlim = c(-1,1),
main = paste(which.module, "Module-trait relationships"))
}
```

## Barplots of interesting module eigengene expression
## Recalculate ME with color labels
```{r}
MEs1 <- moduleEigengenes(expr = datExpr, colors = moduleColors, softPower = sft$powerEstimate)$eigengenes
rownames(MEs1) <- rownames(stable_samples)

MEs1 <- orderMEs(MEs1, greyLast = TRUE)

MEs1$sample <- rownames(MEs1)
MEs1 <- as.data.frame(MEs1)
MEs1 <- MEs1[,c(115, 1:114)]
MEs1$sample <- factor(MEs1$sample, levels = c("Hw1-3a", "Hw1-3b", "Hw1-3c", "Hw2-3a", "Hw2-3b", "Hw2-3c", "Wt1-3a", "Wt1-3b", "Wt1-3c", "Wt2-3a", "Wt2-3b", "Wt2-3c",
                                              "Hw1-6a", "Hw1-6b", "Hw1-6c", "Hw2-6b", "Hw2-6c", "Wt1-6a", "Wt1-6b", "Wt1-6c", "Wt2-6a", "Wt2-6b", "Wt2-6c"), ordered = TRUE)
#MEs1 <- arrange(MEs1, sample)
```

### Set treatment colors
```{r}
gendensity <- c(rep(200, 6), rep(100, 5), rep(30, 6), rep(10, 6))
gencolors <- c(rep("#CCFFCC",3), rep("#99FFCC",3), rep("#CCFFFF",3), rep("#CCCCFF",3), rep("#CCFFCC",3), rep("#99FFCC",2), rep("#CCFFFF",3), rep("#CCCCFF",3))

condcolors <- c(rep("#FF66FF",12), rep("#0000FF",11))
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_LPS-master/WGCNAmodules_Pdam_ctrl-LPS/figures/ModuleBarplots_LPS.pdf", width = 6.5, height = 4)
par(mfrow = c(1,1)  
    #mar = c(8,2,2,2)
    )
for (module in sigModules)
{
which.module <- module
ME <- MEs1[, which.module]
MEframe <- as.data.frame(cbind(MEs1$sample, ME))
colnames(MEframe) <- c("sample", "ME")
MEframe$ME <- as.numeric(MEframe$ME)
MEframe$sample <- as.factor(MEframe$sample)
MEframe$sample <- factor(MEframe$sample, levels = c("Hw1-3a", "Hw1-3b", "Hw1-3c", "Hw2-3a", "Hw2-3b", "Hw2-3c", "Wt1-3a", "Wt1-3b", "Wt1-3c", "Wt2-3a", "Wt2-3b", "Wt2-3c",
                                              "Hw1-6a", "Hw1-6b", "Hw1-6c", "Hw2-6b", "Hw2-6c", "Wt1-6a", "Wt1-6b", "Wt1-6c", "Wt2-6a", "Wt2-6b", "Wt2-6c"), ordered = TRUE)
g <- ggplot(MEframe, aes(x = sample, y = ME)) +
  geom_col(aes(color = sample, fill = sample)) +
  scale_color_manual(values = condcolors) +
  scale_fill_manual(values = gencolors) +
  ylim(-0.6,0.6) +
  ggtitle(paste(module, "eigengene expression barplot")) +
  ylab("Module Eigengene expression") +
  theme(axis.text.x = element_text(angle = 300), legend.position = "none")
print(g)
}
```

## Heatmaps of interesting module gene expression
```{r}
for (module in sigModules)
{
which.module <- module
ME <- MEs1[, which.module]
par(mfrow = c(1,1), mar = c(1,2,4,1))
plotMat(t(scale(datExpr[, moduleColors==which.module ]) ), nrgcols = 30, rlabels = T, clabels = rownames(stable_samples), rcols = which.module, title = which.module)
}
```

```{r}
geneModuleMembership <- as.data.frame(cor(datExpr, MEs, use = "p"))
geneTraitSignificance <- as.data.frame(cor(datExpr, datTraits$lps, use = "p"))
```

## Output gene trait significance and module interconnectivity scatterplots
```{r}
 datKME <- signedKME(datExpr, MEs)
GS <- as.numeric(cor(datExpr, datTraits$lps, use = "p"))
  GScolor <- numbers2colors(GS, signed = T)
  numberOfColumn <- substring(names(datKME), 2)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_LPS-master/WGCNAmodules_Pdam_ctrl-LPS/figures/ModuleKmEScatters_LPS_neg.pdf", width = 4.5, height = 4.5)
 for (module in negModules) {
  which.module <- (gsub("ME", "", module))
  column <- match(module, numberOfColumn)     
  restModule <- moduleNumbers == which.module
  verboseScatterplot(datKME[restModule, column], GS[restModule], 
                     ylim = c(-1,1),
                     xlim = c(-0.5,1),
                     abline = TRUE,
                     col = "black",
                     abline.color = "blue",
                     xlab = paste("Intra-modular connectivity (kME)", sep = ""), 
                     ylab = paste("LPS Gene Significance"),
                     main = NULL) 
  }
```

```{r}
 for (module in posModules[5]) {
  which.module <- (gsub("ME", "", module))
  column <- match(module, colorOfColumn)     
  restModule <- moduleColors == which.module
  
datKME.GS <- data.frame(cbind(datKME[restModule, column], GS[restModule]))
colnames(datKME.GS) <- c("kME", "GS.LPS")
datKME.GS$GeneID <- Genes[restModule]
hikME <- datKME.GS %>% arrange(desc(GS.LPS)) #%>% filter(kME > 0.68 & GS.LPS > 0.5)
genematch <- match(hikME$GeneID, genefeatures$ID)
hikME$GeneInfo <- genefeatures$GeneInfo[genematch]

g <- ggplot(hikME, aes(GS.LPS, kME)) + geom_point(color = which.module) + 
  xlab(paste("Module membership ", which.module, "module")) + 
  ylab(paste("Gene Significance", "LPS")) +
  ggtitle(paste("kME.", which.module, "vs. GS")) + 
  xlim(-0.5, 1) +
  ylim(-0,1.5) +
  theme_bw() +
  geom_text_repel(data = hikME[1:5, ], aes(label = hikME[1:5, ]$GeneInfo), color = "black", size = 2.5, box.padding = unit(0.35, "lines"))
print (g)
}
```
```{r}
write.csv(hikME[1:21, 2:4], file = "~/computing/scripts/EAPSI_LPS-master/WGCNAmodules_Pdam_ctrl-LPS/modules_kME-GS_tables/black_top20.csv")
```


## Module eigengene and gene-scaled profile plots
```{r}
#gather what we know about the genes
geneInfo<-data.frame(row.names = colnames(datExpr), module = moduleNumbers, membership = NA)
for(gene in rownames(geneInfo)){
	currentmod <- geneInfo[gene, 'module']
	geneInfo[gene, 'membership'] <- geneModuleMembership[gene, paste('ME', currentmod, sep='')]
}
geneInfo <- geneInfo[order(geneInfo$module, geneInfo$membership, decreasing=T), ]
genematch <- match(rownames(geneInfo), genefeatures$ID)
geneInfo$Info <- genefeatures$GeneInfo[genematch]
```
```{r}
#plot eigengenes along with highly correlated individual genes
pdf(file = "~/computing/scripts/EAPSI_LPS-master/WGCNAmodules_Pdam_ctrl-LPS/figures/ModuleProfiles_LPS_4.25_pos.pdf", width = 4.5, height = 3.5)

#initialize plot with eigengene -- we scale the expression value to mean=0 sd=1 so we can compare profiles of genes with different absolute expression levels
for (posmodule in posModules){
MEagg <- aggregate(scale(MEs[,posmodule])~Treatment+Colony, FUN = mean, data = stable_samples) %>% arrange(Treatment)
MEagg$TC <- paste(MEagg$Colony, MEagg$Treatment, sep=".")

plot(MEagg$V1, ylim = c(-2,2), type = "l", lwd = 3, col = "black", ylab = "Scaled Expression", xlab = NULL, main = gsub("ME","M",posmodule))
abline(h = 0, lty = 2, lwd = 0.1)
#plot each individual gene with high membership on top of the eigengene
for(gene in rownames(geneInfo)[geneInfo$module==(gsub("ME", "", posmodule)) & geneInfo$membership > 0.8]){
	agg <- aggregate(scale(datExpr[,gene])~Treatment+Colony, FUN = mean, data = stable_samples) %>% arrange(Treatment)
	lines(agg$V1, type = "l", lwd = 0.1, col="black")
}
#replot eigengenes on top of the individual gene expression profiles
lines(MEagg$V1, type = "l", lwd=5,col="red")
}
```

## Output interesting module VisANT edge file
```{r}
#need to save/generate TOM
TOM <- TOMsimilarityFromExpr(datExpr, power = 8)
#setwd("~/computing/scripts/EAPSI-Wanglitung/WGCNAmodules/visANT")

# Select modules (only interested in one for now)
visANTModules <- sigModules
# Select module probes
Genes <- colnames(datExpr)
```

## Import gene name annotation information
```{r}
genenames <- read.delim(file = "/Users/mikeconnelly/computing/sequences/genomes/coral/pocillopora/pdam/pdam_genome_ID_Names.gff", header = F)
colnames(genenames) <- c("ID", "GeneName")
genenames <- genenames %>% filter(genenames$ID %in% genefeatures$ID)
rownames(genenames) <- genenames$ID
```

```{r}
for (visANTmod in visANTModules)
{
inModule <- (moduleNumbers == gsub("ME", "", visANTmod))
# Select the genes within each module and extract gene info
modGeneIDs <- Genes[inModule]
# Select the highly connected genes 
#conmodGeneIDs <- rownames(geneInfo)[geneInfo$module==(gsub("ME", "", visANTmod)) & geneInfo$membership > 0.8]
modGeneNames <- genenames$GeneName[match(modGeneIDs, rownames(genenames))]
# Select the corresponding Topological Overlap
modTOM = TOM[inModule, inModule]
dimnames(modTOM) = list(modGeneIDs, modGeneIDs)
# Export the network into an edge list file VisANT can read
vis = exportNetworkToVisANT(modTOM,
file = paste("~/computing/scripts/EAPSI_LPS-master", "VisANTInput-", gsub("E", "", visANTmod), ".txt", sep=""),
weighted = TRUE,
threshold = 0,
probeToGene = data.frame(modGeneIDs, modGeneNames))
}
```
```{r}
# Restrict the network to the genes with connectivity (kME) greater than 0.8

for (visANTmod in visANTModules)
{
inModule <- (moduleNumbers == gsub("ME", "", visANTmod))
# Select the genes within each module and extract gene info
modGeneIDs <- Genes[inModule]
# Select the highly connected genes 
conmodGeneIDs <- rownames(geneInfo)[geneInfo$module==(gsub("ME", "", visANTmod)) & geneInfo$membership > 0.8]
modGeneNames <- genenames$GeneName[match(modGeneIDs, rownames(genenames))]
conmodGeneNames <- genenames$GeneName[match(conmodGeneIDs, rownames(genenames))]
# Select the corresponding Topological Overlap
modTOM = TOM[inModule, inModule]
dimnames(modTOM) = list(modGeneIDs, modGeneIDs)
# Export the network into an edge list file VisANT can read
vis = exportNetworkToVisANT(modTOM[conmodGeneIDs,conmodGeneIDs],
file = paste("~/computing/scripts/EAPSI_LPS-master/WGCNAmodules_Pdam_ctrl-LPS/visANT/", "VisANTInput-", gsub("E", "", visANTmod), ".txt", sep=""),
weighted = TRUE,
threshold = 0,
probeToGene = data.frame(conmodGeneIDs, conmodGeneNames))
}
```