---
title: "LPS_topGO_DESeq2_Pdam"
author: "Mike Connelly"
date: "10/24/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/projects/EAPSI_Pocillopora_LPS")
```
## Setup packages and working directories
```{r packages, error=FALSE, warning=FALSE, message=FALSE}
library("tidyverse")
library("genefilter")
library("topGO")
library("GO.db")
library("Rgraphviz")
```
## Input required *P. damicornis* GO annotation data and construct Gene-to-GO object for custom annotation mapping
```{r, warning=FALSE}
geneID_GO <- readMappings(file = "./data/pdam_genome_genesGO.txt")
str(head(geneID_GO))
```
## Construct GO-to-Gene object for GO term searches of interesting genes
```{r}
GO_geneID <- inverseList(geneID_GO)
str(head(GO_geneID))
```
## Generate gene universe and GO universe from Gene-to-GO and GO-to-Gene objects
```{r}
geneNames <- names(geneID_GO)
str(head(geneNames))
```
```{r}
GONames <- names(GO_geneID)
str(head(GONames))
```

## Input lists of positively and negatively differentially expressed genes after DESeq2 analysis
LPS_Control (Up/Down)

## Construct topGOdata object using *P. damicornis* gene models, differentially expressed gene lists and gene ontology annotations
## Perform Fisher's exact test to identify enriched gene ontology terms
## Perform Kolmogorov-Smirnov test to identify enriched gene ontology terms
## Analyse highly-significant GO terms and corresponding p-values, then produce a table of results
```{r}
DElists <- list.files(path = "./outputs/DESeq-results/lists", pattern = "_ID.txt", full.names = TRUE)
```
```{r}
topGOanalyzeDE <- function(filename) {
### Custom function for full topGO analysis and results table generation
DEgenes <- read.delim(filename, header = FALSE, stringsAsFactors = FALSE)
DEgenes <- as.data.frame(DEgenes)
geneList <- factor(as.integer(geneNames %in% DEgenes$V1))
names(geneList) <- geneNames
#node size is used to prune the GO hierarchy from the terms which have less than that many annotated genes 
Pdam_GOdata <- new("topGOdata",
              description = DElist,
              ontology = "BP",
              allGenes = geneList,
              nodeSize = 10,
              annot = annFUN.gene2GO,
              gene2GO = geneID_GO)
resultFisher01 <- runTest(Pdam_GOdata, algorithm = "weight01", statistic = "fisher")
resultFisher <- runTest(Pdam_GOdata, algorithm = "classic", statistic = "fisher")
resultKS <- runTest(Pdam_GOdata, algorithm = "classic", statistic = "ks")
resultKSelim <- runTest(Pdam_GOdata, algorithm = "elim", statistic = "ks")
allRes <- GenTable(Pdam_GOdata, 
                   elimKS = resultKSelim,
                   classicKS = resultKS,
                   classicFisher = resultFisher,
                   weight01Fisher = resultFisher01,
                   orderBy = "weight01Fisher", 
                   ranksOf = "weight01Fisher", 
                   topNodes = 50)
print(allRes)
resultfile <- paste(filename, "_topGO.csv", sep = "")
write.csv(allRes, file = resultfile)
}
```
```{r}
topGOanalyze <- function(list) {
### Custom function for full topGO analysis and results table generation
DEgenes <- as.data.frame(list)
geneList <- factor(as.integer(geneNames %in% DEgenes$list))
names(geneList) <- geneNames
#node size is used to prune the GO hierarchy from the terms which have less than that many annotated genes 
Pdam_GOdata <- new("topGOdata",
              #description = "DElist",
              ontology = "BP",
              allGenes = geneList,
              nodeSize = 10,
              annot = annFUN.gene2GO,
              gene2GO = geneID_GO)
resultFisher01 <- runTest(Pdam_GOdata, algorithm = "weight01", statistic = "fisher")
resultFisher <- runTest(Pdam_GOdata, algorithm = "classic", statistic = "fisher")
resultKS <- runTest(Pdam_GOdata, algorithm = "classic", statistic = "ks")
resultKSelim <- runTest(Pdam_GOdata, algorithm = "elim", statistic = "ks")
allRes <- GenTable(Pdam_GOdata, 
                   elimKS = resultKSelim,
                   classicKS = resultKS,
                   classicFisher = resultFisher,
                   weight01Fisher = resultFisher01,
                   orderBy = "weight01Fisher", 
                   ranksOf = "weight01Fisher", 
                   topNodes = 50)
print(allRes)
resultfile <- paste(list, "_topGO.csv", sep = "")
write.csv(allRes, file = resultfile)
}
```

```{r message=TRUE}
for (DElist in DElists)
{
  print(DElist)
  topGOanalyzeDE(DElist)
}
```
```{bash}
mv ./outputs/DESeq-results/lists/*.csv ./outputs/DESeq-results/topGO
bash ./bash/awk_REVIGO_LPS.sh
mv ./outputs/DESeq-results/topGO/*.txt ./outputs/DESeq-results/REVIGO
```
```{bash}
sed -i -e 's/"//g' ~/computing/scripts/EAPSI_LPS-master/DESEqresults_Pdam/REVIGO/*.txt
rm  ~/computing/scripts/EAPSI_LPS-master/DESEqresults_Pdam/REVIGO/*.txt-e
```
```{r}
topGOanalyze(WT2_unique)
```


